py_gtfs_rt_ingestion/lib/ingest.py

```diff
@@ -18,68 +19,46 @@ class NoImplConverter(Converter):
     files.
     """
 
-    def get_tables(self) -> List[Tuple[str, pyarrow.Table]]:
-        return []
-
-    def add_file(self, file: str) -> bool:
-        self.error_files.append(file)
-        return len(self.error_files) > 100
-
-    def reset(self) -> None:
-        self.error_files = []
+    def convert(self) -> None:
+        move_s3_objects(
+            self.files,
+            os.path.join(os.environ["ERROR_BUCKET"], DEFAULT_S3_PREFIX),
+        )
 
-    @property
-    def partition_cols(self) -> list[str]:
-        return []
 
-
-def get_converter(config_type: ConfigType) -> Converter:
+def get_converter(config_type: ConfigType, files: List[str]) -> Converter:
     """
     Get the correct converter object for a given config type
     """
     try:
         if config_type.is_gtfs():
-            return GtfsConverter(config_type)
+            return GtfsConverter(config_type, files)
         if config_type.is_gtfs_rt():
-            return GtfsRtConverter(config_type)
+            return GtfsRtConverter(config_type, files)
     except NoImplException as exception:
         # if we encounter a no impl exception, use a no impl converter that will
         # push all files to the error list
         logging.warning(exception)
 
-    return NoImplConverter(config_type)
+    return NoImplConverter(config_type, files)
 
 
-def ingest_files(files: Iterable[str]) -> Iterable[Converter]:
+def ingest_files(files: List[str]) -> None:
     """
-    Take a bunch of files and sort them into Batches based on their config type
-    (derrived from filename). Each Batch should be under a limit in total
-    filesize.
-
-    :param file: An iterable of filename and filesize tubles to be sorted into
-        Batches. The filename is used to determine config type.
-    :param threshold: upper bounds on how large the sum filesize of a batch can
-        be.
-
-    :yield: Converter object containing a list of tables to write with their
-            prefixes and files to move to error and archive buckets
+    sort the incoming file list by type and create a converter for each type.
+    each converter will ingest and convert its files in its own thread.
     """
-    converters = {t: get_converter(t) for t in ConfigType}
+    grouped_files = group_sort_file_list(files)
 
-    for file in files:
+    converters = []
+    for file_group in grouped_files.values():
```

Comment:
go through each file group, create a converter for that group, and use their convert functions in their own threads.